### ISSUE TITLE
SHIPPING SHOWS INCORRECT RATES !!!! 

### ISSUE THREAD

**Issue by ADVANCEDSPORTVE** [magento/magento2#15439]
### Preconditions

1. 
2. MAGENTO 2.2.2   libmysql - 5.1.73 PHP version: 5.6.30 

### Steps to reproduce

1. GO TO FRONT END OF STORE 
2. CHOOSE ANY ITEM ADD TO CART 
3. ENTER SHIPPING ZIP CODE TO GET RATES 

### Expected result

1. RETAIL CUSTOMER SHOULD ONLY SEE RETAIL SHIPPING COSTS

### Actual result

1. [Screenshot, logs]
CUSTOMERS SEES SHIPPING RATES THAT STORE OWNER PAYS TO SHIPPING COMPANY (FEDEX IN MY CASE) 

ACCORDING TO FEDEX SENIOR TECH WHO LOOKED AT IT, HE SAID THAT THE CURRENT SHIPPING MODULE INSTALLED IS AT LIST 8 YEARS OLD VERSION.. THIS IS A SITUATION THAT IS DRIVING ME CRAZY.. 
1 BECAUSE IT IS DOING THIS WHEN NO ONE HAS CHANGED ANYTHING IN THE SHIPPING SETTINGS.. SO MAGICALLY BROKE ITSELF!!

**Comment by AirmanAJK** [comment]
From what I've seen, the negotiated FedEx rates are preferred over the list rates in Magento. This can be seen in the following file/class:
\vendor\magento\module-fedex\Model\Carrier.php

In the class, there is an array named $_ratesOrder. It is an ordered list of preferred rate types to pick from. The first valid amount in a FedEx rates response from that list is used as the shipping price. If none were found, it falls back to the first rate returned by FedEx.

If I understand your post, you're saying you don't want to use discounted rates for your account, but instead use the rates that everyone without discounted rates would pay. This is not a configurable option in Magento, but you may simply be able to test out changing the order of the $_ratesOrder variable from:

```
protected $_ratesOrder = [
        'RATED_ACCOUNT_PACKAGE',
        'PAYOR_ACCOUNT_PACKAGE',
        'RATED_ACCOUNT_SHIPMENT',
        'PAYOR_ACCOUNT_SHIPMENT',
        'RATED_LIST_PACKAGE',
        'PAYOR_LIST_PACKAGE',
        'RATED_LIST_SHIPMENT',
        'PAYOR_LIST_SHIPMENT',
    ];
```

to

```
protected $_ratesOrder = [
        'RATED_LIST_PACKAGE',
        'PAYOR_LIST_PACKAGE',
        'RATED_LIST_SHIPMENT',
        'PAYOR_LIST_SHIPMENT',
        'RATED_ACCOUNT_PACKAGE',
        'PAYOR_ACCOUNT_PACKAGE',
        'RATED_ACCOUNT_SHIPMENT',
        'PAYOR_ACCOUNT_SHIPMENT',
    ];
```
You also do have the option of adding a percentage markup to any rates returned by FedEx, so you can consider that as well if you want to cover your own shipping/handling costs.

**Comment by ADVANCEDSPORTVE** [comment]
wow ! thanks!!! i had no idea that magento developers did that way...
come to think of it.. in my old 1.7.0.2 ver this never happened , rates always showed at full retail

**Comment by ADVANCEDSPORTVE** [comment]
by the way , why not just take out this completely out of the equation? would that work?

'RATED_ACCOUNT_PACKAGE',
        'PAYOR_ACCOUNT_PACKAGE',
        'RATED_ACCOUNT_SHIPMENT',
        'PAYOR_ACCOUNT_SHIPMENT',
    ];

**Comment by ADVANCEDSPORTVE** [comment]
OK SO I HAVE TO ASK AGAIN... IF ANYONE KNOW FOR SURE SO I DON'T BREAK THE STORE 
why not just take out this completely out of the equation? would that work?

'RATED_ACCOUNT_PACKAGE',
'PAYOR_ACCOUNT_PACKAGE',
'RATED_ACCOUNT_SHIPMENT',
'PAYOR_ACCOUNT_SHIPMENT',
];

**Comment by AirmanAJK** [comment]
As I said in my previous comment, that array is an order of priorities. If and when no items in the list match any results from a FedEx rates query, Magento will simply use the first rate provided in the collection.

That being said, it seems they put these particular items in this collection for good reason, so it's probably best just to reorder them to your liking. However, I don't see any reason why removing the ones you don't want would be harmful give the fallback behavior it has.

BE ADVISED, you shouldn't directly edit Magento core code, especially since any upgrades to Magento will likely wipe out your changes. It's best to create an extension that will modify behavior to suit your needs (through inheritance, plugins, observers, etc). If you insist on changing a core file directly, proceed with caution and test thoroughly. It's best to do such changes in a development environment first.

**Comment by ADVANCEDSPORTVE** [comment]
thanks and sorry for the "caps lock" on previous message ( i was not yelling) 
so if i understand correctly and based on previous store version i had that never gave me that issue, this new M2 ver, it is actually terrible. if they are deciding what is best for each store owner, that is like saying "well your are not in control of your own life" not good. ! who are they to decide what's best for me? 
i guess i will not be using any updates now....

**Comment by ADVANCEDSPORTVE** [comment]
and here is an example , order comes in from Hawaii Fedex 2nd day , customer paid $55.78 when the actual rate is $68.06 .. is M2 going to pay for packaging materials and handling ? i think not...

**Comment by AirmanAJK** [comment]
That's what the shipping markup feature is for. It's what we and everyone else uses to cover expenses. Using the standard rate in comparison to your negotiated rate is not a good way to try and cover expenses. This is why it's not a feature, but a configurable markup is.

**Comment by ADVANCEDSPORTVE** [comment]
correct, but the problem with that is that it can not be used for all types of shipping because the negotiated discount that FedEx gives for the ground versus samrtpost and express is different .. so in using the markup one still loses and or in smartpost and ground it would be too expensive to the customer .. 
so in my case and in conclusion i will make that modification unless i know for sure that updates are automatic? i think they are not, so it should work for me...

**Comment by AirmanAJK** [comment]
It sounds like a better solution for you would be a percentage markup restricted by a minimum and maximum value. For example, the markup needs to be at least $5, but no more than $50, and anything in between based on a percentage is left alone. This can be done with an extension, but out of the box Magento only supports fixed or percentage markups.

**Comment by ADVANCEDSPORTVE** [comment]
i am not going to do markups.. i got some help and we did what you first mentioned.. now shows correct rates.. 
in any case something else even more weird is happening now.. 
Customer placed an order , choosing a PO rather than CC or PayPal.. order shows up and no payment of course .. i called the customer and he assured me that he paid with Credit card... so...... what is going on ? i have no clue since my merchant service assured me there is no transaction

**Comment by ghost** [comment]
Hi @ADVANCEDSPORTVE can you show screenshot, which information retail customer  don't must see?

**Comment by ghost** [comment]
@ADVANCEDSPORTVE  , we are closing this issue due to inactivity. If you'd like to update it, please reopen the issue.

**Comment by raisukesh** [comment]
We implemented FedEx shipping(magento 2.1) , but the value is not showing the account discounted price. currently, a value is showing AED 53 for 2.2kg, but actual value should be AED 33. Please find the below script

 protected $_ratesOrder = [
        'RATED_ACCOUNT_PACKAGE',
        'PAYOR_ACCOUNT_PACKAGE',
        'RATED_ACCOUNT_SHIPMENT',
        'PAYOR_ACCOUNT_SHIPMENT',
        'RATED_LIST_PACKAGE',
        'PAYOR_LIST_PACKAGE',
        'RATED_LIST_SHIPMENT',
        'PAYOR_LIST_SHIPMENT',
    ];

is there any changes needs to be done to display an Account discounted price?

**Comment by raisukesh** [comment]
> From what I've seen, the negotiated FedEx rates are preferred over the list rates in Magento. This can be seen in the following file/class:
> \vendor\magento\module-fedex\Model\Carrier.php
> 
> In the class, there is an array named $_ratesOrder. It is an ordered list of preferred rate types to pick from. The first valid amount in a FedEx rates response from that list is used as the shipping price. If none were found, it falls back to the first rate returned by FedEx.
> 
> If I understand your post, you're saying you don't want to use discounted rates for your account, but instead use the rates that everyone without discounted rates would pay. This is not a configurable option in Magento, but you may simply be able to test out changing the order of the $_ratesOrder variable from:
> 
> ```
> protected $_ratesOrder = [
>         'RATED_ACCOUNT_PACKAGE',
>         'PAYOR_ACCOUNT_PACKAGE',
>         'RATED_ACCOUNT_SHIPMENT',
>         'PAYOR_ACCOUNT_SHIPMENT',
>         'RATED_LIST_PACKAGE',
>         'PAYOR_LIST_PACKAGE',
>         'RATED_LIST_SHIPMENT',
>         'PAYOR_LIST_SHIPMENT',
>     ];
> ```
> to
> 
> ```
> protected $_ratesOrder = [
>         'RATED_LIST_PACKAGE',
>         'PAYOR_LIST_PACKAGE',
>         'RATED_LIST_SHIPMENT',
>         'PAYOR_LIST_SHIPMENT',
>         'RATED_ACCOUNT_PACKAGE',
>         'PAYOR_ACCOUNT_PACKAGE',
>         'RATED_ACCOUNT_SHIPMENT',
>         'PAYOR_ACCOUNT_SHIPMENT',
>     ];
> ```
> You also do have the option of adding a percentage markup to any rates returned by FedEx, so you can consider that as well if you want to cover your own shipping/handling costs.

We implemented FedEx shipping(magento 2.1) , but the value is not showing the account discounted price. currently, a value is showing AED 53 for 2.2kg, but actual value should be AED 33. Please find the below script

protected $_ratesOrder = [
'RATED_ACCOUNT_PACKAGE',
'PAYOR_ACCOUNT_PACKAGE',
'RATED_ACCOUNT_SHIPMENT',
'PAYOR_ACCOUNT_SHIPMENT',
'RATED_LIST_PACKAGE',
'PAYOR_LIST_PACKAGE',
'RATED_LIST_SHIPMENT',
'PAYOR_LIST_SHIPMENT',
];

is there any changes needs to be done to display an Account discounted price?
